File: apps/minecraft/spigot/src/main/java/crabcraft/net/crabUtilities/jade/protocol/provider/ServerExtensionProvider.java
package crabcraft.net.crabUtilities.jade.protocol.provider;

import crabcraft.net.crabUtilities.jade.protocol.accessor.Accessor;
import crabcraft.net.crabUtilities.jade.protocol.util.ViewGroup;

import java.util.List;

public interface ServerExtensionProvider<T> extends JadeProvider {
    List<ViewGroup<T>> getGroups(Accessor<?> request);
}

File: apps/minecraft/spigot/src/main/kotlin/crabcraft/net/crabUtilities/jade/protocol/provider/ServerExtensionProvider.kt
package crabcraft.net.crabUtilities.jade.protocol.provider

import crabcraft.net.crabUtilities.jade.protocol.accessor.Accessor
import crabcraft.net.crabUtilities.jade.protocol.util.ViewGroup

interface ServerExtensionProvider<T> : JadeProvider {
    fun getGroups(request: Accessor<*>): List<ViewGroup<T>>?
}
